### ISSUE TITLE
Fix PyInstaller

### ISSUE THREAD

**Issue by banasiakadam60** [golemfactory/clay#1247]
Currently `PyInstaller` doesn't work on Windows and on MacOS. 
On Windows there are two main problems:

1. building `golemcli` failed with error
```
C:\golem>python -m PyInstaller golemcli.spec
265 WARNING: Internal error: early pywin32 import was introduced
343 INFO: PyInstaller: 3.3.dev0+b20133197
343 INFO: Python: 3.5.3
343 INFO: Platform: Windows-10-10.0.14393-SP0
343 INFO: UPX is not available.
Traceback (most recent call last):
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\site-packages\PyInstaller\__main__.py", line 99, in <module>
    run()
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\site-packages\PyInstaller\__main__.py", line 92, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\site-packages\PyInstaller\__main__.py", line 46, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\site-packages\PyInstaller\building\build_main.py", line 790, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "C:\Users\IEUser\AppData\Local\Programs\Python\Python35\lib\site-packages\PyInstaller\building\build_main.py", line 736, in build
    exec(text, spec_namespace)
  File "<string>", line 9, in <module>
AttributeError: module 'os' has no attribute 'getcwdu'
```

2. Launching `golemapp` failed with an error `Failed to execute script pyi_rth_qt5plugins`. Maybe check

- https://github.com/pyinstaller/pyinstaller/issues/2301

- https://github.com/pyinstaller/pyinstaller/issues/2604

- https://github.com/pyinstaller/pyinstaller/issues/2280

**Comment by banasiakadam60** [comment]
It looks like `pyinstaller` doesn't link to any libraries in `%PYTHONPATH/lib/site-packages` (eg PyQt5). Probably we need to manually point to this dependencies.
See https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs

**Comment by banasiakadam60** [comment]
It looks like we're at the same point on MacOS and Windows now. PyQt is included (in a really nasty way, which have to be changed in the feature, but it works for now)
It would be really, really, really great if someone could try to build it on Windows and check what happens

**Comment by Elfoniok** [comment]
After a week of struggle looks like executable for windows 10 can be build. Although it needs more tests.
Two important things here. 
Looks like python 3.5 is compiled on windows with newer c runtime and it was necessary to recompile pywin32 with same runtime.

As for pyinstaller it still has a lot o problems with python3. However recent pull from github alowes us to build something stable.

As think we have more problems with Mac Os right now.

**Comment by banasiakadam60** [comment]
OK, as for now, it looks like I've fixed Windows problems in #1244 and #1272. I've checked that and also @Radagast-red did. We needed to change some dependencies (eg. change `vc_redist`) and add some switches to the `PyInstaleller`.
On MacOS it looks that @mfranciszkiewicz fixed issues. Everything is merged, so IMO we can close this issue.
